test/dialing/resolver.spec.js

```diff
@@ -0,0 +1,131 @@
+'use strict'
+/* eslint-env mocha */
+
+const { expect } = require('aegir/utils/chai')
+const sinon = require('sinon')
+
+const multiaddr = require('multiaddr')
+const { Resolver } = require('multiaddr/src/resolvers/dns')
+
+const peerUtils = require('../utils/creators/peer')
+const baseOptions = require('../utils/base-options.browser')
+
+const { MULTIADDRS_WEBSOCKETS } = require('../fixtures/browser')
+const relayAddr = MULTIADDRS_WEBSOCKETS[0]
+
+const getDnsaddrStub = (peerId) => [
+  [`dnsaddr=/dnsaddr/ams-1.bootstrap.libp2p.io/p2p/${peerId}`],
+  [`dnsaddr=/dnsaddr/ams-2.bootstrap.libp2p.io/p2p/${peerId}`],
+  [`dnsaddr=/dnsaddr/lon-1.bootstrap.libp2p.io/p2p/${peerId}`],
+  [`dnsaddr=/dnsaddr/nrt-1.bootstrap.libp2p.io/p2p/${peerId}`],
+  [`dnsaddr=/dnsaddr/nyc-1.bootstrap.libp2p.io/p2p/${peerId}`],
+  [`dnsaddr=/dnsaddr/sfo-2.bootstrap.libp2p.io/p2p/${peerId}`]
+]
+
+const relayedAddr = (peerId) => `${relayAddr}/p2p-circuit/p2p/${peerId}`
+
+const getDnsRelayedAddrStub = (peerId) => [
+  [`dnsaddr=${relayedAddr(peerId)}`]
+]
+
+describe('Dialing (resolvable addresses)', () => {
```

Comment:
Add a test where the resolve throws for and address to ensure we can still succeed if others pass.